cmd/mtaBuild.go

```diff
@@ -0,0 +1,411 @@
+package cmd
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/SAP/jenkins-library/pkg/command"
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+	"time"
+)
+
+const templateMtaYml = `_schema-version: "2.0.0"
+ID: "{{.ID}}"
+version: {{.Version}}
+
+parameters:
+  hcp-deployer-version: "1.0.0"
+
+modules:
+  - name: {{.Name}}
+    type: html5
+    path: .
+    parameters:
+       version: {{.Version}}-${timestamp}
+    build-parameters:
+      builder: grunt
+      build-result: dist`
+
+// MTABuildTarget ...
+type MTABuildTarget int
+
+const (
+	// NEO ...
+	NEO MTABuildTarget = iota
+	// CF ...
+	CF MTABuildTarget = iota
+	//XSA ...
+	XSA MTABuildTarget = iota
+)
+
+// ValueOfBuildTarget ...
+func ValueOfBuildTarget(str string) (MTABuildTarget, error) {
+	switch str {
+	case "NEO":
+		return NEO, nil
+	case "CF":
+		return CF, nil
+	case "XSA":
+		return XSA, nil
+	default:
+		return -1, fmt.Errorf("Unknown BuildTarget: '%s'", str)
+	}
+}
+
+// String ...
+func (m MTABuildTarget) String() string {
+	return [...]string{
+		"NEO",
+		"CF",
+		"XSA",
+	}[m]
+}
+
+func mtaBuild(config mtaBuildOptions, telemetryData *telemetry.CustomData, commonPipelineEnvironment *mtaBuildCommonPipelineEnvironment) {
+	log.Entry().Info("Launching mta build")
+	err := runMtaBuild(config, commonPipelineEnvironment, &command.Command{})
+	if err != nil {
+		log.Entry().
+			WithError(err).
+			Fatal("failed to execute mta build")
+	}
+
+}
+
+func runMtaBuild(config mtaBuildOptions, commonPipelineEnvironment *mtaBuildCommonPipelineEnvironment,
+	e envExecRunner) error {
+
+	e.Stdout(os.Stderr) // keep stdout clear.
+	e.Stderr(os.Stderr)
+
+	if len(config.ProjectSettingsFile) > 0 {
```

Comment:
Is the `settings.xml` file just loaded into the workspace and to use it one needs to reference it in the `mta.yml` file?

Reply:
It is not located in the workspace. We have two settings files here:
1.) the global settings file: This is located inside the `$M2_HOME/conf` folder of the maven installation. This is the very last resort if no other settings file is present. Normally this file belongs to the maven installation and should not be overwritten IMO. I don't know the history why our coding is like it is
2.) the users settings file (here named "projectSettingsFile". This is located inside `$HOME/.m2`. This settings file has a higher precedence than the settings file above.

When these files are present they are used without any further need to configure/reference it.

`In general` I think it is a better approach to use the `-s, --settings` flag when invoking maven. Since maven is not invoked here directly, but "transitive" through `mta` it is not that easy to provide that flag. Having this in mind I think working with - at least - one user settings file is ok.